odoo/models.py

```diff
@@ -4062,13 +4080,10 @@ def _filter_access_rules(self, operation):
             return self
 
         # determine ids in database that satisfy ir.rules
-        valid_ids = set()
-        query.add_where(f'"{self._table}".id IN %s')
-        query_str, params = query.select()
         self._flush_search([])
-        for sub_ids in self._cr.split_for_in_conditions(self.ids):
-            self._cr.execute(query_str, params + [sub_ids])
-            valid_ids.update(row[0] for row in self._cr.fetchall())
+        query.add_where(SQL("%s.id IN %s", SQL.identifier(self._table), tuple(self.ids)))
```

Comment:
-> `SQL("%s IN %s", SQL.identifier(self._table, 'id'), tuple(self.ids))`

Reply:
Yes, but it actually changes the generated SQL :confused: 
I wanted to avoid fixing `assertQueries()`, but your suggestion is better, indeed.